crates/types/src/config/common.rs

```diff
@@ -84,8 +88,11 @@ pub struct CommonOptions {
     /// NOTE: This config entry only impacts the initial number of partitions, the
     /// value of this entry is ignored for bootstrapped nodes/clusters.
     ///
+    /// NOTE: Setting this number to 0 will not create any partitions and the system
+    /// will wait until manually provisioned by the command line
+    ///
     /// Cannot be higher than `65535` (You should almost never need as many partitions anyway)
-    pub bootstrap_num_partitions: NonZeroU16,
+    pub bootstrap_num_partitions: u16,
```

Comment:
I am a bit confused now. Unless I misunderstood, last thing we agreed on was to have `profiles` where they only change the default configuration the node starts with (or dumped) which means we still need a way to configure (no partitions) and I think this is better than the `auto_provision_partitions` flag.

I see 2 paths here:
- make num partitions NonZero, and put back the auto_provision_partitions. 
- make num partitions u16, and drop the auto_provision_partitions.

In both cases, the `--production` profile will only tailor the values to sane values suitable for the required operation mode.

Reply:
Would it work if `--production` would disable the `--allow-bootstrap` flag? Then we wouldn't write any `NodesConfiguration`, `PartitionTable` and `LogsConfiguration`. Only once the user calls something like `restatectl cluster provision` this would happen. This would effectively mean that there is no longer a way to provision a cluster with 0 partitions, which I believe is not strictly needed if we have the extra provision step.